services/core/functions/review-lambda/config.ts

```diff
@@ -4,15 +4,12 @@ import { Architecture, Runtime } from "aws-cdk-lib/aws-lambda";
 import { NodejsFunction } from "aws-cdk-lib/aws-lambda-nodejs";
 import { Construct } from "constructs";
 import { join } from "path";
-import { buildResourceName } from "../../helpers/env-helpers";
 
 const OPENAI_API_KEY_PARAM_NAME = "GLOBAL_OPENAI_API_KEY";
 
-export class ReviewLambda extends Construct {
+export class ReviewLambda extends NodejsFunction {
   constructor(scope: Construct, id: string) {
-    super(scope, id);
-
-    const lambda = new NodejsFunction(this, buildResourceName("review-lambda"), {
+    super(scope, id, {
```

Comment:
Out of interest sake why are we doing this?

Reply:
So the aim of this is to reduce "layers of complexity", or deepness.

When you look at cloudformation, the constructs work as a tree, where a construct has child constructs etc.
If the ReviewLambda extends Construct, and then creates a NodejsFunction, this is what it ends up looking like:
<img width="747" alt="image" src="https://github.com/mattzcarey/code-review-gpt/assets/74537705/2108c7b2-cc9c-4633-b49a-d6802b04c9ba">
The "review-lambda-construct" is just used to create "liza-review-lambda", it has no other purpose, so it creates unnecessary complexity. Also the name of the lambda will be created with the id of all parent constructs, and the stack name. So here it ends up being `liza-crgpt-core-stack-reviewlambdaconstructlizarev-AeYWR8JtUP56`, it actually gets cut because there is a max limit, but if it wasn't it would get called `liza-crgpt-core-stack-reviewlambdaconstructlizareviewlambda-AeYWR8JtUP56`, which is quite ugly and repeats information mutiple times

When we make the ReviewLambda extend NodejsFunction instead, and don't use buildResourceName, this is what it ends up looking like:
<img width="747" alt="image" src="https://github.com/mattzcarey/code-review-gpt/assets/74537705/d395e60f-3c76-4525-996e-f4919a70cd6b">
You don't have the unnecessary step anymore, and the lambda name ends up being `manonf-crgpt-core-stack-reviewlambda84CEF8A7-lXugA41uqELT` which feels more explicit

Also when extending NodejsFunciton, if in stack.ts you do `const reviewLambda = new ReviewLamdba(...)`, you get all of the lambda properties and methods directly on reviewLambda, which you don't when extending construct.

So yeah it's because in my different projects, I've noticed that it's often better to extend the relevant construct directly, rather than extend the generic `Construct`, unless you actually need to extend `Construct` because you're doing more of an L3 type thing ;)